### PR TITLE
Add mulitcluster check command

### LINKED ISSUE #5208: Move `check --multicluster` to `multicluster check`
As we modularize linkerd so that the core `install|upgrade|check` is agnostic of extensions (like multicluster & [viz](https://github.com/linkerd/linkerd2/issues/5197)), we should remove the `check --multicluster` flag (which is somewhat difficult to discover) and instead make this command `linkerd multicluster check`.

### PR DESCRIPTION
**Pull request by kleimkuhler**
## What

This change moves the `linkerd check --multicluster` functionality under it's
own multicluster subcommand: `linkerd multicluster check`.

There should be no functional changes as a result of this change. `linkerd
check` no longer checks for anything multicluster related and the
`--multicluster` flag has been removed.

## Why

Closes #5208

The bulk of these changes are moving all the multicluster checks from
`pkg/healthcheck` into the multicluster package.

Doing this completely separates it from core Linkerd. It still uses
`pkg/healtcheck` when possible, but anything that is used only by `multicluster
check` has been moved.

**Note the the `kubernetes-api` and `linkerd-existence` checks are run.**

These checks are required for setting up the Linkerd health checker. They set
the health checker's `kubeAPI`, `linkerdConfig`, and `apiClient` fields.

These could be set manually so that the only check the user sees is
`linkerd-multicluster`, but I chose not to do this.

If any of the setting functions errors, it would just tell the user to run
`linkerd check` and ensure the installation is correct. I find the user error
handling to be better by including these required checks since they should be
run in the first place.

## How to test

Installing Linkerd and multicluster should result in a basic check output:

```
$ bin/linkerd install |kubectl apply -f -
..
$ bin/linkerd check
..
$ bin/linkerd multicluster install |kubectl apply -f -
..
$ bin/linkerd multicluster check
kubernetes-api
--------------
√ can initialize the client
√ can query the Kubernetes API

linkerd-existence
-----------------
√ 'linkerd-config' config map exists
√ heartbeat ServiceAccount exist
√ control plane replica sets are ready
√ no unschedulable pods
√ controller pod is running
√ can initialize the client
√ can query the control plane API

linkerd-multicluster
--------------------
√ Link CRD exists


Status check results are √
```

After linking a cluster:

```
$ bin/linkerd multicluster check
kubernetes-api
--------------
√ can initialize the client
√ can query the Kubernetes API

linkerd-existence
-----------------
√ 'linkerd-config' config map exists
√ heartbeat ServiceAccount exist
√ control plane replica sets are ready
√ no unschedulable pods
√ controller pod is running
√ can initialize the client
√ can query the control plane API

linkerd-multicluster
--------------------
√ Link CRD exists
√ Link resources are valid
        * k3d-y
√ remote cluster access credentials are valid
        * k3d-y
√ clusters share trust anchors
        * k3d-y
√ service mirror controller has required permissions
        * k3d-y
√ service mirror controllers are running
        * k3d-y
× all gateway mirrors are healthy
        probe-gateway-k3d-y.linkerd-multicluster mirrored from cluster [k3d-y] has no endpoints
    see https://linkerd.io/checks/#l5d-multicluster-gateways-endpoints for hints

Status check results are ×
```

Signed-off-by: Kevin Leimkuhler <kevin@kleimkuhler.com>

### CODE REVIEW

**Review by alpeb — COMMENTED** [review]
Looks and tests good :+1: 
Just a couple minor comments.

File: pkg/k8s/api.go [hunk 1]
```diff
@@ -165,6 +165,20 @@ func (kubeAPI *KubernetesAPI) GetReplicaSets(ctx context.Context, namespace stri
...
+		return nil, err
+	}
+	for _, ns := range namespaces.Items {
+		if ns.Labels[LinkerdExtensionLabel] == value {
+			return &ns, err
+		}
+	}
+	return nil, fmt.Errorf("could not find the %s extension", value)
+}
+
```
[the remark this answers is not in the document]
Comment (adleong) [reply pkg/k8s/api.go:181]: I could take this or leave it.  Not a lot of point of changing the error unless we're testing for that.  We can always change the error later if we need to differentiate them.

**Review by adleong — APPROVED** [review]
:+1: to @alpeb's suggestion but otherwise LGTM

File: multicluster/cmd/check.go [hunk 2]
```diff
@@ -1,403 +1,315 @@
...
+	}
+	linkerdHC := healthcheck.NewHealthChecker(checks, &healthcheck.Options{
+		ControlPlaneNamespace: controlPlaneNamespace,
+		KubeConfig:            kubeconfigPath,
+		KubeContext:           kubeContext,
+		Impersonate:           impersonate,
+		ImpersonateGroup:      impersonateGroup,
+		APIAddr:               apiAddr,
+		RetryDeadline:         time.Now().Add(options.wait),
+		MultiCluster:          false,
```
[the remark this answers is not in the document]
Comment (kleimkuhler) [reply multicluster/cmd/check.go:117]: Oh good catch I missed this, thanks!

File: pkg/k8s/api.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (kleimkuhler) [reply pkg/k8s/api.go:181]: I lean towards keeping this as is for now. A reason for that is I'm not sure the status error `NotFound` totally makes sense here. The namespace could in fact exist, but it just may not have the label we are looking for.

I think until we need to make a change and actually use that error somehow, we'll have a better idea of what error status makes sense here and can change it then.

**Review by alpeb — APPROVED** [review]
LGTM :+1:

**Pull request merged**